app/views/groups/index.html.erb

```diff
@@ -1,12 +1,63 @@
-<h1>Groups</h1>
+<div class="container">
+  <div class="my-5">
+    <div class="d-flex flex-row-reverse">
+      <button type="button" class="btn btn-secondary" data-bs-toggle="modal" data-bs-target="#createGroupModal">
+        <%= t(:create_group) %>
+      </button>
+    </div>
+  </div>
+</div>
+<!-- new group modal -->
+<div class="modal fade" id="createGroupModal" tabindex="-1" aria-labelledby="createGroupModalLabel" aria-hidden="true">
+  <div class="modal-dialog modal-dialog-centered">
+    <div class="modal-content">
+      <div class="modal-header border-0">
+        <h5 class="modal-title" id="createGroupModalLabel"><%= t(:create_group) %></h5>
+        <button type="button" class="btn-close" data-bs-dismiss="modal" aria-label="<%= t(:cancel) %>"></button>
+      </div>
+      <%= form_with(model: Group.new) do |form| %>
+        <div class="modal-body">
+          <div>
+            <%= form.label :name, t(:group_name_prompt), class: "form-label" %>
+            <%= form.text_field :name, placeholder: t(:group_name), class: "form-control" %>
+          </div>
+        </div>
+        <div class="modal-footer border-0 justify-content-evenly">
+          <button type="button" class="btn btn-primary" data-bs-dismiss="modal"><%= t(:cancel) %></button>
+          <%= form.submit t(:create_group), :class => "btn btn-secondary" %>
+        </div>
+      <% end %>
+    </div>
+  </div>
+</div>
 
-<div id="groups">
+<%# maybe use https://www.codeply.com/p/0CWffz76Q9 ?? #%>
```

Comment:
Should this comment stay in our code?

Reply:
Probably not, forgot to remove it after resolving the dropdown ux issue with PO